src/core/unit.c

```diff
@@ -975,9 +973,18 @@ void unit_dump(Unit *u, FILE *f, const char *prefix) {
                 prefix, yes_no(u->perpetual),
                 prefix, strna(unit_slice_name(u)),
                 prefix, strna(u->cgroup_path),
-                prefix, yes_no(u->cgroup_realized),
-                prefix, u->cgroup_realized_mask,
-                prefix, u->cgroup_members_mask);
+                prefix, yes_no(u->cgroup_realized));
+
+        if (u->cgroup_realized_mask) {
+                _cleanup_free_ char *s = NULL;
+                cg_mask_to_string(u->cgroup_realized_mask, &s);
+                fprintf(f, "%s\tCGroup mask: %s\n", prefix, s);
```

Comment:
given that this cg_mask_to_string() can fail, in which case s is NULL, we should enclose the usage of "s" in the fprintf() here in an "strnull" or so. After all fprintf is free to not check for NULL, and gcc actually rewrites fprintf as fputs() if the format string follows specific criteria, in which case passing NULL as a string will explode